student/src/main/java/tds/student/services/remote/RemoteOpportunityService.java

```diff
@@ -0,0 +1,141 @@
+package tds.student.services.remote;
+
+import TDS.Shared.Browser.BrowserInfo;
+import TDS.Shared.Exceptions.ReturnStatusException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+import tds.common.Response;
+import tds.common.ValidationError;
+import tds.exam.Exam;
+import tds.exam.OpenExamRequest;
+import tds.student.services.abstractions.IOpportunityService;
+import tds.student.services.data.ApprovalInfo;
+import tds.student.sql.abstractions.ExamRepository;
+import tds.student.sql.data.OpportunityInfo;
+import tds.student.sql.data.OpportunityInstance;
+import tds.student.sql.data.OpportunitySegment;
+import tds.student.sql.data.OpportunityStatus;
+import tds.student.sql.data.OpportunityStatusChange;
+import tds.student.sql.data.OpportunityStatusExtensions;
+import tds.student.sql.data.TestConfig;
+import tds.student.sql.data.TestSegment;
+import tds.student.sql.data.TestSelection;
+import tds.student.sql.data.TestSession;
+import tds.student.sql.data.Testee;
+
+@Service("integrationOpportunityService")
+@Scope("prototype")
+public class RemoteOpportunityService implements IOpportunityService {
+  private final IOpportunityService legacyOpportunityService;
+  private final boolean isRemoteExamCallsEnabled;
+  private final ExamRepository examRepository;
+
+  @Autowired
+  public RemoteOpportunityService(
+    @Qualifier("legacyOpportunityService") IOpportunityService legacyOpportunityService,
+    @Value("${tds.exam.remote.enabled}") Boolean remoteExamCallsEnabled,
+    ExamRepository examRepository) {
+    this.isRemoteExamCallsEnabled = remoteExamCallsEnabled;
+    this.legacyOpportunityService = legacyOpportunityService;
+    this.examRepository = examRepository;
+  }
+
+  @Override
+  public List<TestSelection> getEligibleTests(Testee testee, TestSession session, String grade, BrowserInfo browserInfo) throws ReturnStatusException {
+    return legacyOpportunityService.getEligibleTests(testee, session, grade, browserInfo);
+  }
+
+  @Override
+  public OpportunityInfo openTest(Testee testee, TestSession session, String testKey) throws ReturnStatusException {
+    OpportunityInfo legacyOpportunityInfo = legacyOpportunityService.openTest(testee, session, testKey);
+
+    //This isn't ideal, but due to the way the progman properties are loaded within the system this lives within the service rather than the callers.
```

Comment:
I'll remove this comment.  It was here before I figured out how to get the legacy service to be within this class.  This was solved via the prototype comment above.